src/openms/include/OpenMS/KERNEL/RangeManager.h

```diff
@@ -387,6 +396,28 @@ namespace OpenMS
       return RangeBase::contains(value);
     }
 
+    /* If the current range is a single point (e.g. min==max), then extend the range by min_span/2 on either side.
+     * If @p ppm is True, range min_span is assumed to be in ppm
+     * Unit of min_span is in ppm
+     * @note range min_ cannot be negative, will autocorrect to 0 if have a negative value
+     */
+    void minSpanIfSingular(const double initial_min_span, const bool ppm=false)
+    {
+      if (ppm & (min_ == max_) )
+      {
+        min_ -= min_ * initial_min_span / 2e6;
+        max_ += max_ * initial_min_span / 2e6;
+      }
+      else
+      {
+        RangeBase::minSpanIfSingular(initial_min_span);
+      }
+      if (min_ < 0)
+      {
+        min_ = 0;
+      }
+    }
+
```

Comment:
```suggestion
```

This function must go!
It has side effects and should not be part of RangeManager in general

Reply:
Sorry did not see comment above

Reply:
> Is there another place where this function could be or should I just implement this in OpenSwath classes? It seems like something more generally applicable than OpenSwath though
> 
> > This function must go! It has side effects and should not be part of RangeManager in general

You could create a new `RangeUtils.h` and put common conversion functions in there, e.g. the one from above:
```
  RangeMobility fromIsolationWindowDriftTime(const Precursor& pc);

  RangeMZ fromIsolationWindowMZ(const Precursor& pc);
```

Reply:
it seems that there already is a RangeUtils.h and it does not seem super related. For now I've created a function in DIAHelpers which seems more appropriate for this specific usage